lfs/batcher.go

```diff
@@ -51,16 +55,22 @@ func (b *Batcher) acceptInput() {
 	exit := false
 
 	for {
-		batch := b.newBatch()
+		batch := make([]Transferable, 0, b.batchSize)
 	Loop:
 		for len(batch) < b.batchSize {
-			t, ok := <-b.input
-			if !ok {
-				exit = true // input channel was closed by Exit()
-				break Loop
+			select {
+			case t, ok := <-b.input:
+				if ok {
+					batch = append(batch, t)
+				} else {
+					exit = true // input channel was closed by Exit()
+					break Loop
+				}
+			case <-time.After(time.Millisecond * 300):
```

Comment:
Is it worth introducing a Flush() operation that does this & allows further items to be added but without needing it to be based on a timeout? The timeout is cool in that it's automatic, but it also makes it less deterministic - unrelated changes can alter the behaviour just because of how the times work out. I've seen this problem a lot in debugging (stuff happens differently when you're debugging because different amounts of time pass when you stick on a breakpoint) particularly in real-time systems and it often needs a 'pausable' time source to eliminate. Although debugging is not really a practical problem right now due to the lack of decent Go debugging facilities, differences in server response times could cause difficult to repeat behaviour - some of that is unavoidable but I'd prefer a deterministic sequence unrelated to time, or at least without any parallel time component other than server responses, in our own code if that's possible.